### PR TITLE
fix: fix agent type after ready

### PR DESCRIPTION
**Pull request by gxcsoccer**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

agent 和 app 共用 options，agent 被污染

### CODE REVIEW

File: test/agent.test.js [hunk 1]
```diff
@@ -45,4 +45,10 @@ describe('test/agent.test.js', () => {
       });
     });
   });
+
+  it('should agent work ok after ready', function* () {
+    app = mm.app({ baseDir });
+    yield app.ready();
+    assert(app._agent.type === 'agent');
```
Comment (gxcsoccer) [inline test/agent.test.js]: ready 以后 app._agent.type 会被覆盖为 `application`

File: lib/app.js [hunk 2]
```diff
@@ -59,7 +59,7 @@ class MockApplication extends EventEmitter {
     const egg = require(this.options.framework);
 
     const Agent = egg.Agent;
-    const agent = this._agent = new Agent(this.options);
+    const agent = this._agent = new Agent(Object.assign({}, this.options));
```
Comment (popomore) [inline lib/app.js]: app 也顺便 assign 下？

**Comment by popomore** [comment]
Windows 应该是上次 @dead-horse 那个问题

File: lib/app.js (hunk 2 above)
Comment (gxcsoccer) [inline lib/app.js]: ok

**Comment by gxcsoccer** [comment]
改了，ci 过了发版本吧

**Comment by popomore** [comment]
+ egg-mock@3.9.1

**Pull request merged**